htsohm/material_files.py

```diff
@@ -197,131 +179,95 @@ def write_child_definition_files(run_id, parent_id, generation, mutation_strengt
-    write_force_field(force_field_file)                        # WRITE FORCE_FIELD.DEF
-    print('  - force_field.def\tWRITTEN.')
+    parent_row = session.query(Material).get(str(parent_id))
+    parent_yaml = os.path.join(material_dir, '%s.yaml' % parent_row.uuid)
+    with open(parent_yaml) as load_file:
+        parent_material = yaml.load(load_file)
 
     ########################################################################
-    # copy pseudo_atoms.def
-    parent_forcefield_directory = os.path.join(ff_dir, run_id + '-' + str(parent.uuid))
-    parent_pseudo_file = os.path.join(parent_forcefield_directory, 'pseudo_atoms.def')
-    shutil.copy(parent_pseudo_file, child_forcefield_directory)    # COPY PSEUDO_ATOMS.DEF
-    print('  - pseudo_atoms.def\tWRITTEN.')
+    # add row to database
+    child_row = Material(run_id)
+    child_row.parent_id = parent_id
+    child_row.generation = generation
+    child_material = PseudoMaterial(new_db_row.uuid)
 
     ########################################################################
-    # perturb LJ-parameters, write force_field_mixing_rules.def
-    p_mix = os.path.join(parent_forcefield_directory, 'force_field_mixing_rules.def')
-    n1, n2, old_epsilons, old_sigmas = np.genfromtxt(
-        p_mix, unpack=True, skip_header=7, skip_footer=11
-    )
-    chemical_ids = np.genfromtxt(
-        p_mix, unpack=True, skip_header=7, skip_footer=11, usecols=0, dtype=str
-    )
-    print('Mixing rules\nchem\tep\t\tsig')
-    for i in range(len(chemical_ids)):
-        print('%s\t%s\t%s' % (chemical_ids[i], old_epsilons[i], old_sigmas[i]))
-    atom_types = []
-    for i in range(len(chemical_ids)):
-        epsilon = round( old_epsilons[i] + mutation_strength * (uniform(*epsilon_limits) -
-        old_epsilons[i]), 4)
-        sigma = round(
-            old_sigmas[i] + mutation_strength * (uniform(*sigma_limits) - old_sigmas[i]
-        ), 4)
-        atom_type = {
-            "chemical-id" : chemical_ids[i],
-            "charge"      : 0.,    # See Docstring Todo.
-            "epsilon"     : epsilon,
-            "sigma"       : sigma}
-        atom_types.append(atom_type)
-
-        mix_file = os.path.join(child_forcefield_directory, 'force_field_mixing_rules.def')
-        write_mixing_rules(mix_file, atom_types)
-        print('  - force_field_mixing_rules.def\tWRITTEN.')
+    # perturb LJ-parameters
+    child_material.atom_types = []
+    for atom_type in parent_material.atom_types:    
+        new_atom_type = {'chemical-id' : atom_type['chemical-id']}
+        new_atom_type['epsilon'] = (round(atom_type['epsilon'] +
+                mutation_strength * (uniform(*epsilon_limits) -
+                atom_type['epsilon']), 4))
+        new_atom_type['sigma'] = (round(atom_type['sigma'] +
+                mutation_strength * (uniform(*epsilon_limits) -
+                atom_type['epsilon']), 4))
+        child_material.atom_types.append(new_atom_type)
```

Comment:
Implemented - note: this change fixes a bug addressed in a later commit (PR mutate_refactor).